src/pages/lessons/[slug].tsx

```diff
@@ -495,195 +542,161 @@ const Lesson: React.FC<LessonProps> = ({initialLesson}) => {
-                      lesson,
-                    ).then((lessonView: any) => {
-                      if (lessonView) {
-                        console.debug('progress recorded', {
-                          progress: lessonView,
-                        })
-                        setLessonView(lessonView)
-                      }
+      <div className={cx({'h-screen': isFullscreen})}>
+        <div
+          className={cx(
+            'bg-black w-full lg:grid lg:grid-cols-12 lg:space-y-0',
+            {
+              'absolute top-0': isFullscreen,
+              relative: !isFullscreen,
+            },
+          )}
+          ref={fullscreenWrapperRef}
+        >
+          <div
+            className={cx(
+              'relative before:float-left after:clear-both after:table',
+              {
+                'col-span-9': withSidePanel,
+                'col-span-12': !withSidePanel,
+              },
+            )}
+          >
+            {mounted && (
+              <div className={cx({hidden: !playerVisible})}>
+                <Player
+                  className="font-sans"
+                  container={fullscreenWrapperRef.current || undefined}
+                >
+                  <HLSSource src={lesson.hls_url} />
+                  {lesson.subtitles_url && (
+                    <track
+                      key={`${lesson.slug}-subtitles`}
+                      src={lesson?.subtitles_url}
+                      kind="subtitles"
+                      srcLang="en"
+                      label="English"
+                    />
+                  )}
+                  {metadataTracks && (
+                    <track
+                      key={`${lesson.slug}-metadata`}
+                      id="notes"
+                      src={`/api/lessons/notes/${lesson.slug}?staff_notes_url=${lesson.staff_notes_url}`}
+                      kind="metadata"
+                      label="notes"
+                    />
+                  )}
+                </Player>
+              </div>
+            )}
+            {lessonState.matches('joining') && (
+              <OverlayWrapper>
+                <EmailCaptureCtaOverlay
+                  lesson={lesson}
+                  technology={primary_tag}
+                />
+              </OverlayWrapper>
+            )}
+            {lessonState.matches('subscribing') && (
+              <OverlayWrapper>
+                <GoProCtaOverlay
+                  lesson={lesson}
+                  viewLesson={() => {
+                    send({
+                      type: 'LOAD',
+                      lesson: initialLesson,
+                      viewer,
                     })
+                    // TODO: Make sure this is working as expected
+                    videoService.send({
+                      type: 'LOAD_RESOURCE',
+                      resource: initialLesson as any,
+                    })
+                  }}
+                />
```

Comment:
I couldn't figure out how to test the `viewLesson` handler properly since it's tied to commerce logic. 

@jbranchaud I was wondering if you could please give me a hand testing this? :)

Reply:
A couple things:

1. I just ran this branch locally and was able to play a video, use the controls, etc. I'm not sure if I should look for specific things, but it passed a quick smoke test.

2. The Hard Way™ to be able to test this is to get the Rails server running so that it can manage whether you're allowed to see the resource or not. I'm not sure whether that is a path you want to go down. The setup can be pretty involved.

3. I think when I was first working on this feature, @DrShpongle was porting some of the logic elsewhere and figured out how to use the dev tools to put XState in the state that he wanted for testing. @DrShpongle, any pointers on doing that?

4. If you can describe to me more specifically what you're trying to test and where you are hung up, I can try to give more specific suggestions. Or we could hop on a call sometime next week to walkthrough the parts you want to test together.

Let me know 😃